uploadedfile/thumbfile.go

```diff
@@ -17,172 +17,94 @@ var (
 )
 
 type ThumbFile struct {
-	name        string
-	width       int
-	maxWidth    int
-	height      int
-	maxHeight   int
-	shape       string
-	cropGravity string
-	cropWidth   int
-	cropHeight  int
-	cropRatio   string
-	quality     int
-	format      string
-	localPath   string
-	storeURI    string
-}
-
-func NewThumbFile(width, maxWidth, height, maxHeight int, name, shape, path, cropGravity string, cropWidth, cropHeight int, cropRatio string, quality int) *ThumbFile {
+	Name          string
+	Width         int
+	MaxWidth      int
+	Height        int
+	MaxHeight     int
+	Shape         string
+	CropGravity   string
+	CropWidth     int
+	CropHeight    int
+	CropRatio     string
+	Quality       int
+	Format        string
+	LocalPath     string
+	StoreURI      string
+	DesiredFormat string
+}
+
+func NewThumbFile(width, maxWidth, height, maxHeight int, name, shape, path, cropGravity string, cropWidth, cropHeight int, cropRatio string, quality int, desiredFormat string) *ThumbFile {
 	if quality == 0 {
 		quality = defaultQuality
 	}
 
 	return &ThumbFile{
-		name,
-		width,
-		maxWidth,
-		height,
-		maxHeight,
-		shape,
-		cropGravity,
-		cropWidth,
-		cropHeight,
-		cropRatio,
-		quality,
-		"",
-		path,
-		"",
+		Name:          name,
+		Width:         width,
+		MaxWidth:      maxWidth,
+		Height:        height,
+		MaxHeight:     maxHeight,
+		Shape:         shape,
+		CropGravity:   cropGravity,
+		CropWidth:     cropWidth,
+		CropHeight:    cropHeight,
+		CropRatio:     cropRatio,
+		Quality:       quality,
+		Format:        "",
+		LocalPath:     path,
+		StoreURI:      "",
+		DesiredFormat: desiredFormat,
 	}
 }
 
-func (this *ThumbFile) GetName() string {
-	return this.name
-}
-
-func (this *ThumbFile) SetName(name string) {
-	this.name = name
-}
-
-func (this *ThumbFile) GetHeight() int {
-	return this.height
-}
-
-func (this *ThumbFile) SetHeight(h int) {
-	this.height = h
-}
-
-func (this *ThumbFile) GetMaxHeight() int {
-	return this.maxHeight
-}
-
-func (this *ThumbFile) SetMaxHeight(h int) {
-	this.maxHeight = h
-}
-
-func (this *ThumbFile) GetWidth() int {
-	return this.width
-}
-
-func (this *ThumbFile) SetWidth(h int) {
-	this.width = h
-}
-
-func (this *ThumbFile) GetMaxWidth() int {
-	return this.maxWidth
-}
-
-func (this *ThumbFile) SetMaxWidth(h int) {
-	this.maxWidth = h
-}
-
-func (this *ThumbFile) GetShape() string {
-	return this.shape
-}
-
-func (this *ThumbFile) SetShape(shape string) {
-	this.shape = shape
-}
-
-func (this *ThumbFile) GetPath() string {
-	return this.localPath
-}
-
 func (this *ThumbFile) SetPath(path string) error {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		return errors.New(fmt.Sprintf("Error when creating thumbnail %s", this.GetName()))
+		return errors.New(fmt.Sprintf("Error when creating thumbnail %s", this.Name))
 	}
 
-	this.localPath = path
+	this.LocalPath = path
```

Comment:
Perhaps `LocalPath` shouldn't be exported since we always want to run it this existence check